dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/teradata/TeradataLogsConnector.java

```diff
@@ -251,10 +249,33 @@ public void addTasksTo(List<? super Task<?>> out, @Nonnull ConnectorArguments ar
     if (!isAssessment) {
       conditionsBuilder.add("L.UserName <> 'DBC'");
     }
+
+    SharedState queryLogsState = new SharedState();
+    SharedState utilityLogsState = new SharedState();
     ImmutableSet<String> conditions = conditionsBuilder.build();
-    for (ZonedInterval interval : intervals) {
-      String file = createFilename(ZIP_ENTRY_PREFIX, interval);
-      if (isAssessment) {
+
+    if (isAssessment) {
+      String utilityLogsTable =
+          arguments.getDefinitionOrDefault(TeradataLogsConnectorProperty.UTILITY_LOGS_TABLE);
+
+      { // fail fast validation step
```

Comment:
How about moving the code to a method `failFastValidationStep` (or similar) instead of a comment and a code block?

Reply:
Done